airflow/pipe_events_fishing_dag.py

```diff
@@ -0,0 +1,46 @@
+from datetime import datetime, timedelta
+
+from airflow import DAG
+from airflow.operators.bash_operator import BashOperator
+
+from pipe_tools.airflow.models import DagFactory
+
+
+class PipelineDagFactory(DagFactory):
+    def source_date_range(self):
+        # The scored messages only have logistic scores for a couple of days
+        # while we accumulate the amount of data we need to refine that score
+        # with the nnet model. We need to shift the intervals' start date so
+        # that it reprocesses the events for up to 7 days before the current
+        # day.
+        if self.schedule_interval == '@daily':
+            return '{{ ds + macros.dateutil.relativedelta.relativedelta(days=-7) }}', '{{ ds }}'
+        elif self.schedule_interval == '@monthly':
+            start_date = '{{ (execution_date.replace(day=1) + macros.dateutil.relativedelta.relativedelta(days=-7)).strftime("%Y-%m-%d") }}'
+            end_date = '{{ (execution_date.replace(day=1) + macros.dateutil.relativedelta.relativedelta(months=1)).strftime("%Y-%m-%d") }}'
```

Comment:
@andres-arana shouldn't the end date also be substracting the last 7 days?  If not, if I am reading the code correctly for example March 2018 you would be taking from `Feb 21` to `March 31`. Am I wrong?

Reply:
Not really, the problem is that the currently scored messages have the logistic network scores, so you can process whatever is already in the scored messages table. However, you need to re-process the data up to 7 days before the start of the boundary, so that you re-process the days that might have changed due to the new nnet scores being set.